Objects/listobject.c

```diff
@@ -233,6 +233,75 @@ valid_index(Py_ssize_t i, Py_ssize_t limit)
     return (size_t) i < (size_t) limit;
 }
 
+#ifdef Py_GIL_DISABLED
+
+static PyObject *
+list_item_impl(PyListObject *self, Py_ssize_t idx, PyObject *dead)
```

Comment:
The `dead` argument here is not being used. The idea was to keep the `Py_DECREF` calls off the fast path, but it probably doesn't help right now so I'd suggest just removing "dead".